crates/mod-rspamd/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+    #[serde(default)]
+    reject_spam: bool,
+}
+
+fn default_true() -> bool {
+    true
+}
+
+impl RspamdClientConfig {
+    /// Build rspamd-client Config from this configuration
+    async fn make_client_config(&self) -> anyhow::Result<Config> {
+        // Extract password if provided
+        let password = if let Some(p) = &self.password {
+            let bytes = p.get().await?;
+            Some(String::from_utf8(bytes.to_vec()).context("password is not valid UTF-8")?)
+        } else {
+            None
+        };
+
+        // Extract encryption key if provided
+        let encryption_key = if let Some(k) = &self.encryption_key {
+            let bytes = k.get().await?;
+            Some(String::from_utf8(bytes.to_vec()).context("encryption_key is not valid UTF-8")?)
+        } else {
+            None
+        };
+
+        // Prepare optional proxy config
+        let proxy_config = if let Some(proxy_url) = &self.proxy_url {
+            let username = if let Some(u) = &self.proxy_username {
+                let bytes = u.get().await?;
+                Some(String::from_utf8(bytes.to_vec()).context("proxy_username is not valid UTF-8")?)
+            } else {
+                None
+            };
+
+            let proxy_password = if let Some(p) = &self.proxy_password {
+                let bytes = p.get().await?;
+                Some(String::from_utf8(bytes.to_vec()).context("proxy_password is not valid UTF-8")?)
+            } else {
+                None
+            };
+
+            Some(rspamd_client::config::ProxyConfig {
+                proxy_url: proxy_url.clone(),
+                username,
+                password: proxy_password,
+            })
+        } else {
+            None
+        };
+
+        // Construct Config directly since all fields are public
+        // Note: use HTTPCrypt encryption_key for secure communication
+        Ok(Config {
+            base_url: self.base_url.to_string(),
+            password,
+            timeout: self.timeout.map(|t| t.as_secs_f64()).unwrap_or(30.0),
+            retries: self.retries.unwrap_or(1),
+            tls_settings: None, // use httpcrypt instead of tls
+            proxy_config,
+            zstd: self.zstd,
+            encryption_key,
+        })
+    }
+}
+
+/// Wrapper around rspamd-client Config for Lua
+#[derive(Clone)]
+struct RspamdClient {
+    config: Arc<Config>,
+}
+
+impl RspamdClient {
+    async fn new(lua_config: RspamdClientConfig) -> anyhow::Result<Self> {
+        let config = lua_config.make_client_config().await?;
+        Ok(Self {
+            config: Arc::new(config),
+        })
+    }
+
+    fn get_config(&self) -> mlua::Result<&Config> {
+        Ok(&self.config)
+    }
+
+    async fn scan(
+        &self,
+        message: String,
+        envelope: EnvelopeDataLua,
+    ) -> anyhow::Result<RspamdScanReply> {
+        let config = self.get_config().map_err(|e| anyhow::anyhow!("{}", e))?;
```

Comment:
nit: prefer this form of format string expansion, as it is less prone to drifting in unexpected ways across future refactoring!

```suggestion
        let config = self.get_config().map_err(|e| anyhow::anyhow!("{e}"))?;
```